cocotb/regression.py

```diff
@@ -319,6 +325,12 @@ def _result_was():
             self.log.info("Test errored as expected: " + _result_was())
 
         else:
+            if _pdb_on_exception:
+                if sys.version_info >= (3, 5):
+                    traceback = exc_info.__traceback__
+                else:
+                    traceback = exc_info[2]
+                pdb.post_mortem(traceback)
             self.log.error("Test Failed: " + _result_was(), exc_info=exc_info)
```

Comment:
I think it would be helpful for this line to appear before the post-mortem, to remind the user exactly which test they're looking at.